var/spack/repos/builtin/packages/berkeleygw/package.py

```diff
@@ -0,0 +1,159 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+import os
+
+
+class Berkeleygw(MakefilePackage):
+    """BerkeleyGW is a many-body perturbation theory code for excited states,
+    using the GW method and the GW plus Bethe-Salpeter equation (GW-BSE) method
+    to solve respectively for quasiparticle excitations and optical properties of
+    materials."""
+
+    homepage = "https://berkeleygw.org"
+
+    version('2.1',
+            '31f3b643dd937350c3866338321d675d4a1b1f54c730b43ad74ae67e75a9e6f2',
+            url='https://berkeley.box.com/shared/static/ze3azi5vlyw7hpwvl9i5f82kaiid6g0x.gz',
+            expand=False)
+
+    variant('elpa', default=True, description='Build with ELPA support')
+    variant('openmp', default=True, description='Build with OpenMP support')
+
+    depends_on('mpi')
+    depends_on('blas')
+    depends_on('lapack')
+    depends_on('scalapack')
+
+    depends_on('fftw+openmp', when='+openmp')
+    depends_on('fftw~openmp', when='~openmp')
+
+    depends_on('elpa+openmp', when='+elpa+openmp')
+    depends_on('elpa~openmp', when='+elpa~openmp')
+
+    depends_on('hdf5+fortran+hl')
+
+    depends_on('py-h5py', type=('build', 'run'))
+
+    depends_on('perl', type='test')
+
+    parallel = False
+
+    def edit(self, spec, prefix):
+        # archive is a tar file, despite the .gz expension
+        os.system("tar xf %s --strip-components=1" % self.stage.archive_file)
```

Comment:
```suggestion
        tar('-x', '-f', self.stage.archve_file, '--strip-components=1')
```

Reply:
Recommend using Spack's built-in executable for commands.

Reply:
btw, is there a way to override the expand method in `version` instead of `expand=False` and this?

Reply:
Not that I am aware.